lib/mail_handler/reply_handler.rb

```diff
@@ -96,22 +96,50 @@ def self.is_oof?(message)
 
     def self.forward_on(raw_message, message = nil)
       forward_to = self.get_forward_to_address(message)
-      IO.popen("/usr/sbin/sendmail -i #{forward_to}", "wb") do |f|
+      IO.popen(%Q(/usr/sbin/sendmail -i "#{forward_to}"), 'wb') do |f|
         f.write(raw_message);
         f.close;
       end
     end
 
     def self.get_forward_to_address(message)
-      forward_to = AlaveteliConfiguration.forward_nonbounce_responses_to
       if AlaveteliConfiguration.enable_alaveteli_pro
-        pro_contact_email = AlaveteliConfiguration.pro_contact_email
-        original_to = message ? MailHandler.get_all_addresses(message) : []
-        if original_to.include?(pro_contact_email)
-          forward_to = AlaveteliConfiguration.forward_pro_nonbounce_responses_to
-        end
+        non_default_recipient =
+          if addressed_to_both_contacts?(message)
+            [default_recipient, pro_recipient].join(',')
+          elsif addressed_to_pro_contact?(message)
+            pro_recipient
+          end
+
+        return non_default_recipient if non_default_recipient
       end
-      forward_to
+
+      default_recipient
+    end
```

Comment:
I suppose we could make the top-level method (`get_forward_to_address`) a little simpler, at the expense of more class methods passing the `message` around. This should really be a class that encapsulates `message` rather than a module.

```ruby
def self.get_forward_to_address(message)
  non_default_recipient(message) || default_recipient
end

def self.non_default_recipient(message)
  if AlaveteliConfiguration.enable_alaveteli_pro
    if addressed_to_both_contacts?(message)
      [default_recipient, pro_recipient].join(',')
    elsif addressed_to_pro_contact?(message)
      pro_recipient
    end
  end
end

def self.default_recipient
  AlaveteliConfiguration.forward_nonbounce_responses_to
end
```